Dockerfile

```diff
@@ -21,9 +23,9 @@ RUN \
 
 ADD . $APPLOC
 WORKDIR $APPLOC
-RUN go get -d && \
-    go build -o /bin/ceph_exporter
-
+RUN go get -d
+RUN if [ -n "${TEST}" ]; then go test -v ./...; fi
+RUN go build -o /bin/ceph_exporter
```

Comment:
Is it useful to test without generating the binary? I think we can treat it as 2 separate operations. Just have a test hook that doesn't generate any binaries and then one that performs the multi-stage build.

Reply:
That's how I initially planned it but then I though it would cause the initial apt setup to be run twice in a CI pipeline context.